packages/scoutgame/src/builderNfts/recordNftMint.ts

```diff
@@ -213,6 +214,83 @@ export async function recordNftMint(
     log.error('Error completing quest', { error, builderId: builderNft.builderId, questType: 'scout-starter-card' });
   }
 
+  try {
+    // Find the referrer of the scout first
+    const referrer = await prisma.scout.findFirst({
+      where: {
+        events: {
+          some: {
+            referralCodeEvent: {
+              refereeId: scoutId
+            }
+          }
+        }
+      },
+      select: {
+        id: true
+      }
+    });
+
+    if (referrer) {
+      const referralCodeEvent = await prisma.referralCodeEvent.findFirst({
+        where: {
+          refereeId: scoutId
+        },
+        select: {
+          id: true
+        }
+      });
+
+      // If the referral code event exists and doesn't have a bonus event, create one
+      if (referralCodeEvent && !referralCodeEvent.bonusEvent) {
+        await prisma.$transaction(async (tx) => {
+          await tx.builderEvent.create({
+            data: {
+              type: 'referral_bonus',
+              season: currentSeason,
+              description: 'Received points for a referred user scouting a builder',
+              week: getCurrentWeek(),
+              builder: {
+                connect: {
+                  id: referrer.id
```

Comment:
at the moment there's no guarantee that referrer.id is the same as referralCodeEvent.builderEvent.builderId